src/python/grpcio/grpc/experimental/aio/_call.py

```diff
@@ -209,10 +205,13 @@ def time_remaining(self) -> Optional[float]:
         return self._cython_call.time_remaining()
 
     async def initial_metadata(self) -> MetadataType:
-        return await self._cython_call.initial_metadata()
+        raw_metadata_tuple = await self._cython_call.initial_metadata()
+        return Metadata(*(raw_metadata_tuple or ()))
```

Comment:
Should we make `Metadata` take in tuple directly? We could perform some type checking in its `__init__` function. WDYT?

Reply:
+1

And we should revisit the return type, either go for `Optional[MetadatType]` or `MetadataType` but not both of them mixed.

For keeping backward compatibility as much as possible we would need to go for `Optional[MetadatType]`.

Reply:
Alternatively, you could add a staticmethod `Metadata.from_tuple`. Then you wouldn't incur the cost of runtime type checks.